oak_restricted_kernel_wrapper/build.rs

```diff
@@ -24,7 +24,18 @@ fn main() {
     println!("cargo:rerun-if-changed=layout.ld");
     println!("cargo:rustc-link-arg=--script=layout.ld");
     let kernel_directory = "oak_restricted_kernel_bin";
-    let file_name = "oak_restricted_kernel_simple_io_bin";
+    let file_name = match (cfg!(feature = "oak_restricted_kernel_bin"), cfg!(feature = "oak_restricted_kernel_simple_io_bin")) {
+        (true, false) => "oak_restricted_kernel_bin",
+        (false, true) => "oak_restricted_kernel_simple_io_bin",
+        (true, true) => panic!("oak_restricted_kernel_simple_io_bin and feature oak_restricted_kernel_bin cannot be enabled at the same time. Only either version can be built."),
+        (false, false) => panic!("One of oak_restricted_kernel_simple_io_bin and feature oak_restricted_kernel_bin must be enabled.")
+    };
+
+    if cfg!(feature = "oak_restricted_kernel_simple_io_bin") {
+        "oak_restricted_kernel_simple_io_bin"
+    } else {
+        "oak_restricted_kernel_bin"
+    };
```

Comment:
Is this pretty? No. But it gets the job done. 

I don't like that we're repurposing rust features to set which binary to wrap, but it's the most pragmatic way to pass args to the build script. 

We could instead also simply rely on the caller of this crate overwriting the input file with the right version in advance, but it's imo messier.